### PR TITLE
doc: fix broken link for cephfs kernel recommendations

### PR DESCRIPTION
**Pull request by gouthampacha**
Content was moved around in commit c74a261e5b, this
commit updates the name of the file that contains the
kernel recommendations for cephfs users.

Signed-off-by: Goutham Pacha Ravi <gouthampravi@gmail.com>

### CODE REVIEW

**Comment by tchaikov** [comment]
jenkins render docs

File: doc/start/os-recommendations.rst [hunk 1]
```diff
@@ -117,4 +117,4 @@ Testing
 
 .. _CRUSH Tunables: ../../rados/operations/crush-map#tunables
 
-.. _CephFS best practices: ../../cephfs/best-practices
+.. _CephFS best practices: ../../cephfs/mount-using-kernel-driver
```
Comment (tchaikov) [inline doc/start/os-recommendations.rst:120]: in addition to changing the link, i think we also need to change the text to "Mount CephFS using Kernel Driver".

Comment (gouthampacha) [reply doc/start/os-recommendations.rst:121]: Sure, makes sense. Done.

**Comment by tchaikov** [comment]
jenkins render docs

File: doc/start/os-recommendations.rst [hunk 2]
```diff
@@ -117,4 +118,4 @@ Testing
 
 .. _CRUSH Tunables: ../../rados/operations/crush-map#tunables
 
-.. _CephFS best practices: ../../cephfs/best-practices
+.. _Mount CephFS using Kernel Driver: ../../cephfs/mount-using-kernel-driver
```
Comment (rishabh-d-dave) [inline doc/start/os-recommendations.rst:121]:
```suggestion
.. _Mount CephFS using Kernel Driver: ../../cephfs/mount-using-kernel-driver#which-kernel-version
```

Comment (gouthampacha) [reply doc/start/os-recommendations.rst:121]: Cool, makes sense. Done

**Comment by gouthampacha** [comment]
jenkins render docs

**Comment by gouthampacha** [comment]
jenkins render docs

**Comment by tchaikov** [comment]
jenkins test make check

**Comment by gouthampacha** [comment]
jenkins test make check

**Pull request merged**